examples/tutorials/physics_benchmarking.py

```diff
@@ -0,0 +1,334 @@
+    collisions = data["collisions"]
+    observations = data["observations"]
+    end_time = time.time()
+
+    test_log = ""
+    test_log += "Test: %s completed" % test_id + "\n"
+    test_log += "\t Description: " + benchmarks[test_id]["description"] + "\n"
+    test_log += (
+        "\t ========================= Performance ======================== " + "\n"
+    )
+    test_log += "\t| Number of frames: " + str(len(physics_step_times)) + "\n"
+    test_log += (
+        "\t| Average time per physics step: "
+        + str(sum(physics_step_times) / len(physics_step_times))
+        + "\n"
+    )
+    test_log += (
+        "\t| Average time per frame render: "
+        + str(sum(graphics_render_times) / len(graphics_render_times))
+        + "\n"
+    )
+    test_log += "\t| Maximum Collisions: " + str(max(collisions)) + "\n"
+    test_log += "\t| Total time for test: " + str(end_time - start_time) + "\n"
+    test_log += "\t|----- END of %s -----" % test_id + "\n"
+    test_log += (
+        "\t ============================================================== " + "\n"
+    )
+    test_log += "\n"
+
+    print(test_log)
+    if make_video:
+        vut.make_video(
+            observations,
+            "rgba_camera_1stperson",
+            "color",
+            output_path + test_id,
+            open_vid=show_video,
+        )
+
+    for graph_name in graphs_to_produce:
+        if graph_name in data:
+            create_frame_line_plot(test_id, data[graph_name], graph_name)
+
+    return test_log
+
+
+def main():
+    log_contents = ""
+    if len(tests_to_run) == 0:
+        # runs all benchmarks when none are specified
+        for test_id in benchmarks:
+            log_contents += run_test(test_id)
+    else:
+        # only runs the benchmark/benchmark_sets specified in commandline arguments
+        for test in tests_to_run:
+            if test in benchmark_sets:
+                for sub_test in benchmark_sets[test]:
+                    log_contents += run_test(sub_test)
+            elif test in benchmarks:
+                log_contents += run_test(test)
+            else:
+                print(test, "done not exist.")
+                log_contents += test + " does not exist." + "\n" + "\n"
+
+    if save_log:
+        with open(output_path + "benchmark_log_file.txt", "w") as log:
+            log.write(log_contents)
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--no-show-video", dest="show_video", action="store_false")
+    parser.add_argument("--no-make-video", dest="make_video", action="store_false")
+    parser.add_argument("--save-to-log", dest="save_log", action="store_true")
+    parser.add_argument("--graph", default=[], nargs="*", type=str, metavar="N")
+    parser.add_argument("--run", default=[], nargs="*", type=str, metavar="N")
+
+    parser.set_defaults(show_video=True, make_video=True, save_log=False)
+    args, _ = parser.parse_known_args()
```

Comment:
Suggestion: error on unknown args so user gets a clear signal if they messed up? 🤷 
```suggestion
    args = parser.parse_args()
```

Reply:
^ parse_known_args is a required workaround to get it to work in Python notebooks. This isn't compiled as a colab, but maybe it should be?

Reply:
> ^ parse_known_args is a required workaround to get it to work in Python notebooks. This isn't compiled as a colab, but maybe it should be?

We could. Just out of curiosity - are colabs used for anything other than tutorials?

Reply:
@ScottCSteinhauser The real purpose of the Colab is to provide a no-install option to use Habitat-sim in the browser with minimal setup. Tutorials are a good example of why we'd want to do this. 🙂 

The purpose of this script is to benchmark systems, I don't know that we are targeting Colab. I guess an argument could be made that you may want to know physics perf on the Colab node also, but I'd rather keep it clean and simple.